pelican/utils.py

```diff
@@ -999,3 +996,31 @@ def maybe_pluralize(count, singular, plural):
     if count == 1:
         selection = singular
     return '{} {}'.format(count, selection)
+
+
+@contextmanager
+def temporary_locale(temp_locale=None, lc_category=locale.LC_ALL):
+    '''
+    Enable code to run in a context with a temporary locale
+    Resets the locale back when exiting context.
+
+    Use TestCaseWithCLocale if you want every test to use the C locale.
+    '''
+    orig_locale = locale.setlocale(lc_category)
+    if temp_locale:
+        locale.setlocale(lc_category, temp_locale)
+    yield orig_locale
+    locale.setlocale(lc_category, orig_locale)
+
+
+class TestCaseWithCLocale(unittest.TestCase):
```

Comment:
This is a unittest helper rather than a general pelican util and thus should go into `pelican.tests.support`.

Reply:
Moved.